pep-0695.rst

```diff
@@ -875,6 +878,181 @@ incorrectly, because functions within a class do not normally have access to
 the enclosing class namespace. The implementation will retain a reference to
 the class namespace so that class-scoped names are resolved correctly.
 
+.. _695-scoping-behavior:
+
+Scoping Behavior
+----------------
+
+The new syntax requires a new kind of scope that behaves differently
+from existing scopes in Python. Thus, the new syntax cannot be described exactly in terms of
+existing Python scoping behavior. This section specifies these scopes
+further by reference to existing scoping behavior: the new scopes behave
+like function scopes, except for a number of minor differences listed below.
+
+All examples include functions introduced with the pseudo-keyword ``def695``.
+This keyword will not exist in the actual language; it is used to
+clarify that the new scopes are for the most part like function scopes.
+
+``def695`` scopes differ from regular function scopes in the following ways:
+
+- If a ``def965`` scope is immediately within a class scope, or within another
+  ``def695`` scope that is immediately within a class scope, then names defined
+  in that class scope can be accessed within the ``def695`` scope. (Regular functions,
+  by contranst, cannot access names defined within an enclosing class scope.)
+- The following constructs are disallowed anywhere within a ``def695`` scope, though
+  they may be used within regular functions nested inside a ``def695`` scope:
+
+  - ``yield``
+  - ``yield from``
+  - ``await``
+  - ``:=`` (walrus operator)
+
+- The qualified name (``__qualname__``) of objects (classes and functions) defined within ``def695`` scopes
```

Comment:
This would be clearer if you just said that the new scopes do not affect `__qualname__`. In fact, maybe you can omit this altogether. Is there any reason why the reader would think that `__qualname__` is affected?

Reply:
Well, that flows out of the comparison to functions: if these really were functions, then the qualname of a generic function would be something like `__generic_parameters_of_f.<locals>.f`, and we don't want that.

Reply:
It's true that this is a context where the equivalence to function scopes doesn't work well, but overall def695 scopes are still very much like functions; if we defined them in some other way, we'd have a lot more cases to define.